src/stores/exercise.coffee

```diff
@@ -1,169 +1,41 @@
 _ = require 'underscore'
 flux = require 'flux-react'
-
-EXERCISE_MODES_TEMPLATE = [
-  'VIEW'      # just render the exercise (with radios and input boxes)
-  'REVIEW'    # like view but without checkboxes and the correct/incorrect answer selected
-  'EDIT'      # view + all the edit buttons and quill if content clicked
-]
-
-EXERCISE_MODES = {}
-for val, i in EXERCISE_MODES_TEMPLATE
-  EXERCISE_MODES[val] = "ENUM_#{val}_#{i}" # To make sure it's not hardcoded
-
-aryRemove = (ary, item) ->
-  index = ary.indexOf(item)
-  if index >= 0
-    ary.splice(index, 1)
-  else
-    throw new Error('BUG: Item not found in array')
-  true
+{CrudConfig, makeSimpleStore, extendConfig} = require './helpers'
+{QuestionActions, QuestionStore} = require './question'
 
 ExerciseConfig = {
-  # Exercise has a current mode it is rendered as.
-  changeExerciseMode: (newMode) ->
-    @_currentMode = newMode
-    @emitChange()
-
-  changeExerciseStimulus: (exercise, html) ->
-    exercise.stimulus_html = html # TODO: Change to background_html
-    @emitChange()
-
-  removeExerciseStimulus: (exercise) ->
-    # delete exercise.stimulus_html
-    # Ruby doesn't recognize an deleted stimulus so make it blank
-    exercise.stimulus_html = ''
-    @emitChange()
+  _loaded: (obj, exerciseId) ->
+    for question in obj.questions
+      QuestionActions.loaded(question, question.id)
+    obj
 
-  addQuestion: (exercise, question) ->
-    exercise.questions.push(question)
-    @emitChange()
+  updateNumber: (id, number) -> @_change(id, {number})
 
-  removeQuestion: (exercise, question) ->
-    aryRemove(exercise.questions, question)
-    @emitChange()
+  updateStimulus: (id, stimulus_html) -> @_change(id, {stimulus_html})
 
-  moveQuestionUp: (exercise, question) ->
-    i = exercise.questions.indexOf(question)
-    throw new Error('BUG: Invalid position') unless i > 0
-    exercise.questions.splice(i, 1)
-    exercise.questions.splice(i - 1, 0, question)
-    @emitChange()
+  updateTags: (id, tags) -> @_change(id, {tags})
 
-  moveQuestionDown: (exercise, question) ->
-    i = exercise.questions.indexOf(question)
-    unless i < exercise.questions.length - 1
-      throw new Error('BUG: Invalid position')
-    exercise.questions.splice(i, 1)
-    exercise.questions.splice(i + 1, 0, question)
-    @emitChange()
-
-  # Question
-
-  changeQuestion: (question, html) ->
-    question.stem_html = html
-    @emitChange()
-
-  # changeQuestionStimulus: (question, html) ->
-  #   question.stimulus_html = html
-  #   @emitChange()
+  save: (id) ->
+    questions = _.map @_local[id].questions, (question) ->
+      QuestionActions.sync(question.id)
+      QuestionStore.get(question.id)
```

Comment:
Yea, I probably didn't come up with the best name there, but it's meant to grab all the questions and answer json from the QuestionStore and AnswerStore before it saves.